packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -33,32 +42,55 @@ export type InsightsProps<
   onEvent?: (event: InsightsEvent, insightsClient: TInsightsClient) => void;
 };
 
+const ALGOLIA_INSIGHTS_SRC =
+  'https://cdn.jsdelivr.net/npm/search-insights@2.3.0/dist/search-insights.min.js';
+
 export type CreateInsightsMiddleware = typeof createInsightsMiddleware;
 
 export function createInsightsMiddleware<
   TInsightsClient extends null | InsightsClient
->(props: InsightsProps<TInsightsClient>): InternalMiddleware {
+>(props: InsightsProps<TInsightsClient> = {}): InternalMiddleware {
   const {
     insightsClient: _insightsClient,
     insightsInitParams,
     onEvent,
-  } = props || {};
+  } = props;
+
+  let insightsClient: InsightsClient = _insightsClient || noop;
+
   if (_insightsClient !== null && !_insightsClient) {
-    if (__DEV__) {
-      throw new Error(
-        "The `insightsClient` option is required if you want userToken to be automatically set in search calls. If you don't want this behaviour, set it to `null`."
-      );
-    } else {
-      throw new Error(
-        'The `insightsClient` option is required. To disable, set it to `null`.'
-      );
-    }
+    safelyRunOnBrowser(({ window }: { window: any }) => {
+      const pointer = window.AlgoliaAnalyticsObject || 'aa';
+
+      if (typeof pointer === 'string') {
+        insightsClient = window[pointer];
+      }
+
+      if (!insightsClient) {
+        window.AlgoliaAnalyticsObject = pointer;
+        window[pointer] =
+          window[pointer] ||
+          function (...args: any[]) {
+            (window[pointer].queue = window[pointer].queue || []).push(args);
+          };
```

Comment:
Should we test the queue now that we have it implemented on our end?

Maybe:
- The assignment then swap as the library is loaded
- The processing of events sent before loading

Reply:
that would mean we'd have to mock or somehow wait on the code from jsDelivr right? As that's the code which actually processes the queue.

Reply:
Yes. That could be an opportunity to use something like [MSW](https://mswjs.io/docs/) to mock the service being queried rather than write complicated mocks ourselves.

Reply:
to make this work, we'll need to find a way to pass `runScripts: "dangerously"` to JSDOM, I couldn't find a way to do that on a cursory glance yet

Reply:
Let's do this in a follow-up PR. Could be interesting indeed, but I think it may require quite a lot of unrelated changes in configuration etc.